selffuzz/src/test/java/com/code_intelligence/selffuzz/driver/FuzzedDataProviderImplFuzzTest.java

```diff
@@ -0,0 +1,433 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.selffuzz.driver;
+
+import com.code_intelligence.jazzer.junit.FuzzTest;
+import com.code_intelligence.jazzer.mutation.annotation.NotNull;
+import com.code_intelligence.selffuzz.jazzer.api.FuzzedDataProvider;
+import com.code_intelligence.selffuzz.jazzer.driver.FuzzedDataProviderImpl;
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+public class FuzzedDataProviderImplFuzzTest {
+  @FuzzTest
+  void fuzzedDataProviderTest(byte @NotNull[] buf) {
+    try (FuzzedDataProviderImpl data = FuzzedDataProviderImpl.withJavaData(buf)) {
+      ArrayList<Consumer<FuzzedDataProvider>> actionList = getActionList();
+
+      while (data.remainingBytes() > 0) {
+        int actionIndex = data.consumeInt(0, actionList.size() - 1);
+        Consumer<FuzzedDataProvider> action = actionList.get(actionIndex);
+        action.accept(data);
+      }
+    }
+  }
+
+  ArrayList<Consumer<FuzzedDataProvider>> getActionList() {
+    ArrayList<Consumer<FuzzedDataProvider>> actionList = new ArrayList<>();
+    actionList.add(this::testBoolean);
```

Comment:
That's a great list, thanks for the effort!

You can return `Arrays.asList(...)` to make this less verbose, but there is no need to change this now if you don't want to.

Changing the return type to `List<...>` and wrapping the result into `Collections.unmodifiableList(...)` would be welcome though.